var/spack/repos/builtin/packages/mawk/package.py

```diff
@@ -13,5 +13,6 @@ class Mawk(AutotoolsPackage):
     url      = "http://invisible-mirror.net/archives/mawk/mawk-1.3.4.tgz"
 
     version('1.3.4', 'b1d27324ae80302452d0fa0c98447b65')
+    version('1.3.4_p20171017', sha256='ec8cd5b9ab0ea48ec1a0422ecaf20c2f24fcd099d8cc98dcf36eb3617a34bdab', url='https://github.com/ThomasDickey/mawk-snapshots/archive/t20171017.tar.gz')
```

Comment:
Can you sort these versions from newest to oldest? Also, it looks like this tarball is available from invisible-mirror.net, it might be better to download everything from the same source.